bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-                )
-        else:
-            raise BentoMLException(
-                "Unknown type for `model`."
-                f" Got {type(obj)} while only accepted"
-                " one of the following types:"
-                " `Pipeline` and `Union[PreTrainedModel,TFPreTrainedModel,FlaxPreTrainedModel]`"
-            )
-
-        return _model.tag
+    ) as bento_model:
+        pipeline.save_pretrained(bento_model.path)
 
+        return bento_model.tag
 
-# TODO: import_from_huggingface_hub
 
+def get_runnable(
+    bento_model: bentoml.Model,
+) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
 
-class _TransformersRunner(BaseModelRunner):
-    def __init__(
-        self,
-        tag: t.Union[str, Tag],
-        tasks: str,
-        *,
-        framework: str,
-        device: int,
-        name: t.Optional[str] = None,
-        **pipeline_kwargs: t.Any,
-    ):
-        super().__init__(tag=tag, name=name)
-
-        try:
-            transformers.pipelines.check_task(tasks)  # type: ignore
-        except KeyError as e:
-            raise BentoMLException(
-                f"{e}, as `{tasks}` is not recognized by transformers."
-            )
-        self._tasks = tasks
-        self._framework = framework
-        self._device = device
-        self._pipeline_kwargs = pipeline_kwargs
-
-        # pipeline arguments
-        self._feature_extractor = pipeline_kwargs.pop("feature_extractor", None)
-        self._revision = pipeline_kwargs.pop("revision", None)  # type: t.Optional[str]
-        self._use_fast = pipeline_kwargs.pop("use_fast", True)  # type: bool
-        self._use_auth_token = pipeline_kwargs.pop(
-            "use_auth_token", None
-        )  # type: t.Optional[t.Union[str, bool]]
-        self._model_kwargs = pipeline_kwargs.pop(
-            "model_kwargs", {}
-        )  # type: t.Dict[str, t.Any]
-        self._kwargs = pipeline_kwargs
-
-        # tokenizer-related
-        try:
-            self._has_tokenizer = (
-                self.model_info.info.options["feature_extractor"] is False
-            )
-        except Exception:
-            self._has_tokenizer = False
-        self._tokenizer = None
-        self._pipeline = None
-        self._config = None
-
-    @property
-    def num_replica(self) -> int:
-        # TODO: supports multiple GPUS
-        return 1
-
-    def _setup(self) -> None:
-        params = load(
-            self._tag,
-            from_flax="flax" in self._framework,
-            from_tf="tf" in self._framework,
-            return_config=True,
-            revision=self._revision,
-            use_fast=self._use_fast,
-            use_auth_token=self._use_auth_token,
-            model_store=self.model_store,
-            **self._kwargs,
-        )
+    class TransformersRunnable(bentoml.Runnable):
+        SUPPORT_NVIDIA_GPU = True  # type: ignore
+        SUPPORT_CPU_MULTI_THREADING = True  # type: ignore
```

Comment:
we don't want to support multi-threading on huggingface. What do you think about this?

Reply:
Why not? We should always considering multi-threading first

Reply:
Huggingface models are essential TF, PT, or JAX models. All of them support multi-threading, no?